pageserver/src/tenant/timeline.rs

```diff
@@ -1079,7 +1079,16 @@ impl Timeline {
                 if let Err(e) = local_layer.delete() {
                     error!("failed to remove layer file on evict after replacement: {e:#?}");
                 }
-
+                // Always decrement the physical size gauge, even if we failed to delete the file.
+                // Rationale: we already replaced the layer with a remote layer in the layer map,
+                // and any subsequent download_remote_layer will
+                // 1. overwrite the file on disk and
+                // 2. add the downloaded size to the resident size gauge.
+                //
+                // If there is no re-download, and we restart the pageserver, then load_layer_map
+                // will treat the file as a local layer again, count it towards resident size,
+                // and it'll be like the layer removal never happened.
```

Comment:
This is only about eviction, right?
so, we won't end up with local layer that must be deleted somehow resurrecting after pageserver restart?

Reply:
If we fail to delete the file here, it remains in the timeline directory, untracked by the pageserver.
If we restart the pageserver, `load_layer_map` will observe that the file is still there, and add it to the layer map.
It's as if the eviction never happened.

We have the same problem for gc & compaction. @koivunej and I talked this through a while back and we came to the conclusion that, at least for gc, it's fine because gc is idempotent.
Compaction would probably re-process that layer file, but, nothing bad will come from it either.

Question answered? Remaining concerns? Can I word the comment more clearly?

Reply:
Thank you!
I was mostly worried about compaction.
Good to know that it is a known behavior)

Reply:
"known behavior" is probably an overstatement, but yeah.